# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/OffscreenCanvas.cpp
# Commit: ff14d3c64e53
# Full Hash: ff14d3c64e53dad7b4720c983b8f19bb494ed28a
# Author: Ehsan Akhgari <ehsan@mozilla.com>
# Date: 2019-03-06 04:36:55
# Regressor Bug: 1532414
# File Overlap Count: 1
# Description:
#   Bug 1532414 - Pass a document/principal pointer to nsContentUtils::ShouldResistFingerprinting() callers in the canvas API; r=baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22002
# ==============================================================================

diff -r 14b6b6f8a599 -r ff14d3c64e53 dom/canvas/OffscreenCanvas.cpp
--- a/dom/canvas/OffscreenCanvas.cpp	Tue Mar 05 12:42:51 2019 +0000
+++ b/dom/canvas/OffscreenCanvas.cpp	Tue Mar 05 12:42:51 2019 +0000
@@ -255,7 +255,9 @@
   // so that we can decide when usePlaceholder should be true/false?
   // See https://trac.torproject.org/18599
   // For now, we always return a placeholder if fingerprinting resistance is on.
-  bool usePlaceholder = nsContentUtils::ShouldResistFingerprinting();
+  dom::WorkerPrivate* workerPrivate = dom::GetCurrentThreadWorkerPrivate();
+  bool usePlaceholder =
+      nsContentUtils::ShouldResistFingerprinting(workerPrivate->GetPrincipal());
   CanvasRenderingContextHelper::ToBlob(aCx, global, callback, aType, aParams,
                                        usePlaceholder, aRv);
 